tests/tests/test_integration/test_detection.py

```diff
@@ -50,3 +61,36 @@ def test_detection_full():
     assert isclose(
         num_cells_validation, num_cells_test, abs_tol=DETECTION_TOLERANCE
     )
+
+
+def test_callbacks(signal_array, background_array):
+    # 20 is minimum number of planes needed to find > 0 cells
+    signal_array = signal_array[0:20]
+    background_array = background_array[0:20]
+
+    planes_done = []
+    batches_classified = []
+    points_found = []
+
+    def detect_callback(plane):
+        planes_done.append(plane)
+
+    def classify_callback(batch):
+        batches_classified.append(batch)
+
+    def detect_finished_callback(points):
+        points_found.append(points)
+
+    main(
+        signal_array,
+        background_array,
+        voxel_sizes,
+        detect_callback=detect_callback,
+        classify_callback=classify_callback,
+        detect_finished_callback=detect_finished_callback,
+    )
+
+    np.testing.assert_equal(planes_done, np.arange(len(signal_array)))
+    np.testing.assert_equal(batches_classified, [0])
+    assert len(points_found) == 1
+    assert len(points_found[0]) == 120
```

Comment:
could add the appropriate error message which explains what the error is in each of these cases, i.e. `assert x==y, "some explanation here"`

Reply:
Added